### PR TITLE
enh(install): rework install.sh script

### PR DESCRIPTION
**Pull request by cgagnaire**
## Description

This PR aims to provide a hardened installation from sources script.
It includes all manual pre/post installation steps spotted by the GOAT https://github.com/kermith72 on his [blog](https://www.sugarbug.fr/atelier/installations/debian/debian10_centreon_2010/).
It also includes all source files added by our RPM packaging like third-party components configurations.
It is less chatty.
It has been tested on CentOS 7 & 8, Oracle Linux 8, Debian 10.8, Ubuntu 20.10 and openSUSE Leap 15.2.
It still creates the instCentWeb.conf file needed by UI.

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [x] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 19.10.x
- [ ] 20.04.x
- [ ] 20.10.x
- [x] 21.04.x (master)

## How this pull request can be tested?

- Install a Debian, openSUSE or Ubuntu,
- Follow documentation on WIP branch https://github.com/centreon/centreon-documentation/blob/MON-6704/en/installation/installation-using-sources/central-platform.md#centreon-web,
- Get ensure that everything works fine.

## Checklist

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by kduret — DISMISSED** [review]

File: install/src/centreon-apache-https.conf [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+Header set X-Frame-Options: "sameorigin"
+Header always edit Set-Cookie ^(.*)$ $1;HttpOnly;Secure
+ServerSignature Off
+ServerTokens Prod
+TraceEnable Off
+
+Alias /centreon/api @CENTREON_INSTALL_DIR@
+Alias /centreon @CENTREON_INSTALL_DIR@/www/
+
+<LocationMatch ^/centreon/(?!api/latest/|api/beta/|api/v[0-9]+/|api/v[0-9]+\.[0-9]+/)(.*\.php(/.*)?)$>
```
Comment (jeremyjaouen) [inline install/src/centreon-apache-https.conf:13]:
```suggestion
<LocationMatch ^/centreon/(authentication|api/(latest|beta|v[0-9]+|v[0-9]+\.[0-9]+))/.*$>
```
Shouldnt we add the authentication path ?

File: install/src/centreon-apache-https.conf [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    ProxyPassMatch fcgi://127.0.0.1:9042@CENTREON_INSTALL_DIR@/www/$1
+</LocationMatch>
+
+<LocationMatch ^/centreon/api/(latest/|beta/|v[0-9]+/|v[0-9]+\.[0-9]+/)(.*)$>
```
Comment (jeremyjaouen) [inline install/src/centreon-apache-https.conf:17]:
```suggestion
<LocationMatch ^/centreon/(authentication|api/(latest|beta|v[0-9]+|v[0-9]+\.[0-9]+))/.*$>
```
shouldn't we add the authentication path ?

Comment (cgagnaire) [reply install/src/centreon-apache-https.conf:17]: Well I'll let you do it, I don't have permissions :)
Don't forget the other conf.

Comment (lpinsivy) [reply install/src/centreon-apache-https.conf:17]: not yet

**Review by jeremyjaouen — DISMISSED** [review]

File: install/src/centreon-apache-https.conf [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        Require all granted
+        <IfModule mod_php5.c>
+            php_admin_value engine Off
+        </IfModule>
+
+        RewriteRule ^index\.html$ - [L]
+        RewriteCond %{REQUEST_FILENAME} !-f
+        RewriteCond %{REQUEST_FILENAME} !-d
+        RewriteRule . /index.html [L]
+        ErrorDocument 404 /centreon/index.html
```
Comment (sc979) [inline install/src/centreon-apache-https.conf:56]:
```suggestion
        FallbackResource /centreon/index.html
```

File: install/src/centreon-apache.conf [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    Require all granted
+    <IfModule mod_php5.c>
+        php_admin_value engine Off
+    </IfModule>
+
+    RewriteRule ^index\.html$ - [L]
+    RewriteCond %{REQUEST_FILENAME} !-f
+    RewriteCond %{REQUEST_FILENAME} !-d
+    RewriteRule . /index.html [L]
+    ErrorDocument 404 /centreon/index.html
```
Comment (sc979) [inline install/src/centreon-apache.conf:38]:
```suggestion
    FallbackResource /centreon/index.html
```

**Review by sc979 — DISMISSED** [review]
Taking in account PR : https://github.com/centreon/centreon-build/pull/519/files

**Pull request closed without merging**